### ISSUE TITLE
[ft2] should we keep that “font_diskcache”? it’s broken as for now

### ISSUE THREAD

**Issue by NaitLee** [DarkPlacesEngine/darkplaces#66]
The “font diskcache” thing isn’t working: cache are only [stored](https://github.com/DarkPlacesEngine/darkplaces/blob/305eb3834a21c5ceaee8531f645848878ad10152/ft2.c#L1636), but no code is using them. (and they are overwrote again in the next game startup)

In usual scenario we are just gonna fix it. But, by patching #49 (and #65), caching seems unnecessary for increased overall performance; and the fix will become complicated, since map state should be cached too, that’s probably not optimal.

(Another problem of caching: cache path only contains “primary” font, changes to font “groups” will not apply without cleaning them)

So I’d give a proposal: turn this into “diskdump” for developers with new CVar like `r_font_map_dumptexture`, and deprecate the CVar `r_font_diskcache` (leave it ignored)

I will do it along with fixes, if you are affirmative :)

**Comment by NaitLee** [comment]
In my fix a requirement to make `map_identifier` much more unique made diskcaching not feasible.

I should just remove it as for now.

**Comment by NaitLee** [comment]
just a note: the code in `div0-stable` is working, but broken for years in `master`.

**Comment by bones-was-here** [comment]
I don't see any code for loading the cached textures in div0-stable either.  Presumably it could save a bit of CPU time at startup when using `r_font_compress` but it wouldn't improve FPS so probably not worth the implementation time.

I'd just remove it completely unless you know of a use case for “diskdump” feature?

**Comment by NaitLee** [comment]
When working on my patch (#67) I used the "diskcache" to see what's going on with rendering -- that is, helping developers.

If cache won't be used then it's called dump :P 

Anyway, after problems were fixed, dump just also became useless.

The most benefit of font caching is yet with CJK language, but the "incmap" in my patch made caching neither useful nor feasible.

The caching thing is already marked as deprecated in my patch.

Another pitfall of caching is, when font file is updated, the cached textures won't reflect.
